File: 251004/Krotsyuk/module-publisher/src/main/java/org/ikrotsyuk/bsuir/modulepublisher/repository/WriterRepository.java
package org.ikrotsyuk.bsuir.modulepublisher.repository;

import org.ikrotsyuk.bsuir.modulepublisher.entity.WriterEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface WriterRepository extends JpaRepository<WriterEntity, Long> {
    Optional<WriterEntity> findByLogin(String login);

    boolean existsById(Long id);
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/repository/WriterRepository.kt
package bsuir.dc.publisher.repository

import bsuir.dc.publisher.entity.Writer
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WriterRepository : JpaRepository<Writer, Long> {
    fun findByLoginIgnoreCase(login: String): Writer?
}